modules/tinymce/src/core/main/ts/api/EditorManager.ts

```diff
@@ -373,19 +374,18 @@ const EditorManager: EditorManager = {
       ' '
     );
 
-    const isInvalidInlineTarget = function (settings, elm) {
-      return settings.inline && elm.tagName.toLowerCase() in invalidInlineTargets;
-    };
+    const isInvalidInlineTarget = (settings: RawEditorSettings, elm: HTMLElement) =>
+      settings.inline && elm.tagName.toLowerCase() in invalidInlineTargets;
 
-    const createId = function (elm) {
+    const createId = function (elm: HTMLElement) {
       let id = elm.id;
 
       // Use element id, or unique name or generate a unique id
       if (!id) {
-        id = elm.name;
+        id = (elm as unknown as HTMLTextAreaElement).name as string | undefined;
```

Comment:
This is almost definitely the wrong type, as there are multiple elements this could be that have a `name` attribute.

Reply:
I suspect the best solution here would be to change the `elm` type to `elm: HTMLElement & { name?: string }`

Reply:
Another option, which would likely be even better, would be to do something like this:
```
id = Obj.get(elm, 'name').filter((name) => !DOM.get(id)).getOrThunk(DOM.uniqueId);
```